WeatherForecast/WeatherForecast/SceneDelegate.swift

```diff
@@ -15,7 +15,21 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
         // Use this method to optionally configure and attach the UIWindow `window` to the provided UIWindowScene `scene`.
         // If using a storyboard, the `window` property will automatically be initialized and attached to the scene.
         // This delegate does not imply the connecting scene or session are new (see `application:configurationForConnectingSceneSession` instead).
-        guard let _ = (scene as? UIWindowScene) else { return }
+        guard let scene = (scene as? UIWindowScene) else { return }
+        
+        let weatherListViewController: WeatherListViewController = .init(
+            weatherRepository: WeatherRepositoryImpl(dataSource: WeatherDataSourceImpl(), decoder: DataDecoderJSON(decoder: JSONDecoder())),
+            imageFetcher: ImageFetcherImpl()
+        )
+        let navigationController: UINavigationController = UINavigationController(rootViewController: weatherListViewController)
+        navigationController.navigationBar.prefersLargeTitles = true
+        navigationController.navigationBar.tintColor = .black
+        
+        let window: UIWindow = UIWindow(windowScene: scene)
+        window.rootViewController = navigationController
+        
+        self.window = window
+        window.makeKeyAndVisible()
```

Comment:
```suggestion
        let appDependency = AppDependency.resolve(scene)
      self.window = appDependency.window
```
관리할 수도 있을 것 같습니다 ! (하나의 방법이니, 참고만 해주세요 🙇🏻)

Reply:
리뷰 감사합니다!
이렇게 분리하는건 객체미용체조 6원칙을 적용하기 위함일까요? 

<img width="1274" alt="Screenshot 2024-03-20 at 9 15 52 AM" src="https://github.com/yagom-academy/refactoring-weather/assets/38378136/b2894fb5-55ed-4f9a-8054-4303d29b4176">

Reply:
음, 이건 DI 하는 방식중 하나라고 봐주시면 좋을 것 같아요 :)